packages/fuels-core/src/encoding_utils.rs

```diff
@@ -1,17 +1,19 @@
+use crate::constants::ENUM_DISCRIMINANT_WORD_WIDTH;
 use crate::{EnumVariants, ParamType, WORD_SIZE};
 
-// Calculates how many WORDs are needed to encode any given variant in `variants`
-pub fn encoding_width_to_fit_any(variants: &EnumVariants) -> usize {
+// Calculates how many WORDs are needed to encode an enum
+pub fn compute_encoding_width_of_enum(variants: &EnumVariants) -> usize {
     variants
         .param_types()
         .iter()
-        .map(expected_encoding_width)
+        .map(compute_encoding_width)
         .max()
-        .expect("Will never panic because EnumVariants must have at least variant inside it!")
+        .map(|width| width + ENUM_DISCRIMINANT_WORD_WIDTH)
+        .expect("Will never panic because EnumVariants must have at least one variant inside it!")
 }
 
 /// Calculates the number of `WORD`s the VM expects this parameter to be encoded in.
-pub fn expected_encoding_width(param: &ParamType) -> usize {
+pub fn compute_encoding_width(param: &ParamType) -> usize {
```

Comment:
`compute_parameter_type_encoding_width` would be a better name IMO, even if it's the argument, from the outside it would be clearer, and less likely to be confused with the one for enums

Reply:
Rust and your IDE won't allow you to mix these two up. Since the argument type is part of the fn signature, putting it in the function name is redundant.